mllib/src/test/scala/org/apache/spark/ml/classification/LogisticRegressionSuite.scala

```diff
@@ -883,6 +884,22 @@ class LogisticRegressionSuite
     assert(model1a0.intercept ~== model1b.intercept absTol 1E-3)
   }
 
+  test("logistic regression with all labels the same") {
+    val lr = new LogisticRegression()
+      .setFitIntercept(true)
+      .setMaxIter(3)
+    val sameLabels = dataset
+      .withColumn("zeroLabel", lit(0.0))
+      .withColumn("oneLabel", lit(1.0))
+
+    val model = lr
+      .setLabelCol("oneLabel")
+      .fit(sameLabels)
+
+    assert(model.coefficients ~== Vectors.dense(0.0) absTol 1E-3)
+    assert(model.intercept === Double.PositiveInfinity)
```

Comment:
Can you add another test showing that all `zeroLabel` will return intercept with `Double.NegativeInfinity`?

Thanks.